src/Compilers/Core/Portable/Compilation/IStatement.cs

```diff
@@ -70,7 +70,7 @@ public interface ISwitchStatement : IStatement
     /// <summary>
     /// Represents a C# case or VB Case statement.
     /// </summary>
-    public interface ICase
+    public interface ICase : IStatement
```

Comment:
This is because the backing node in bound-tree is a BoundStatement, but I'm not convinced about this either.